DSD-834: consistent naming for props, enums, files

EdwinGuzman opened the pull request:
Fixes JIRA ticket [DSD-834](https://jira.nypl.org/browse/DSD-834)

## This PR does the following:

- This is still a work in progress because we need to talk about the `Hero` and its image-related props, but take a look at the changelog for a full description of all the changes:

```
### Adds

- Created new `LayoutTypes` enum for row and column layouts.

### Removals

- Removes the `CardLayouts`, `CheckboxGroupLayoutTypes`, `RadioGroupLayoutTypes`, and `SkeletonLoaderLayouts` enums.
- Removes the `CardTypes.tsx`,` CheckboxGroupLayoutTypes.tsx`, and `RadioGroupLayoutTypes.tsx` files.

### Updates

- Renames the `Accordion` prop `contentData` to `accordionData`.
- Renames the `Breadcrumbs` prop `colorVariant` to `breadcrumbsType`.
- Renames the `Card` prop `center` to `isCentered`.
- Updates the `Card` prop `layout` to use the `LayoutTypes` enum instead of `CardLayouts`.
- Updates the `CheckboxGroup` prop `layout` to use the `LayoutTypes` enum instead of `CheckboxGroupLayoutTypes`.
- Renames the `Heading` prop `displaySize` to `size.
- Renames the `HeadingDisplaySizes` enum to `HeadingSizes.
- Renames the `Hero` prop `image` to `imageComponent`.
- Renames the `Image` prop `imageSize` to `size`.
- Renames the `Notification` prop `centered` to `isCentered`.
- Updates the `RadioGroup` prop `layout` to use the `LayoutTypes` enum instead of `RadioGroupLayoutTypes`.
- Internal updates to the `SearchBar` component based on updates from the `Select` and `TextInput` components.
- Renames the `Select` prop `type` to `selectType`.
- Updates the `SkeletonLoader` prop `layout` to use the `LayoutTypes` enum instead of `SkeletonLoaderLayouts`.
- Renames the `StructuredContentImage` prop `imageSize` to `size` (this component is based on the `Image` component).
- Internal updates to the `StructuredContent` component based on updates from the `Image` and `Heading` components.
- Renames the `Tabs` prop `contentData` to `tabsData`.
- Renames the `Text` prop `displaySize` to `size`.
- Renames the `TextInput` prop `variant` to `textInputType`.
- Renames the `ToggleSizes.tsx` file to `ToggleTypes.tsx`.
```

It is worth noting that for the `size` prop, even though it's the same name, we cannot use the same enum value. Every component has a slightly different enum object to use for its size:

| component | values |
| ----------- | ------ |
| Toggle       |  Large, Small |
| text | Default, Caption, Tag, Mini |
| progressindicator |  Default, Small |
| logo | ExtraExtraSmall, ExtraSmall, Medium, Large, Default
| icon | Small, Medium, Large, ExtraLarge, ExtraExtraLarge, ExtraExtraExtraLarge
| image | Default, ExtraExtraSmall, ExtraSmall, Small,  Medium, Large
| heading | Primary, Secondary, Tertiary, Callout 


## How has this been tested?

Storybook and tests. No functional updates just naming convention updates.

## Accessibility concerns or updates

<!--- Describe any accessibility concerns or updates that were made that should be known. -->

- N/A

### Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

<!--- This step is done AFTER creating a PR. -->
<!--- Tugboat creates a static Storybook preview URL once the PR is created. -->
<!--- Copy the URL to the relevant Storybook page here. -->

- [ ] View [the example in Storybook]()

EdwinGuzman committed:
WIP trying to make all the props, enums, and file names consistent

EdwinGuzman commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr892-tfew8tb9ymfbizvyvlanhovaj4toozuc.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/622683f501c6d20b85350e27

bigfishdesign13 commented:
@EdwinGuzman 
Let’s try to bring some more consistency to the `size` enum values.
- `Toggle`: change `Large` and `Default`
- `Icon`: add `Default` and have it act the same as `Image` and `Logo`
- `Heading`: should we add `Default` and have it align with the `Secondary` styles?
- Anything that has `ExtraExtra` seems foolish, but can we recalibrate and come up with a better scale?

EdwinGuzman commented:
I updated the `Toggle` to have `Default` and `Small`. The `Icon` component is also updated but they don't all act the same. The default for `Logo` is `Medium` and maybe it should stay that way? Or else the default is 100% width which is huge.

`Heading` has its own logic which includes different h* sizes, so that one may not work or should be refactored.

Yes, but let's make the scaling of enum values a different ticket.

EdwinGuzman committed:
Another round of updates

EdwinGuzman commented:
Marking this as ready to review and latest round of updates:
```
### Adds
- Adds the value "Default" to the `IconSizes` enum.

### Updates
- Updates the `Card` image-related props into one main prop named `imageProps`. This new prop contains the following properties: alt, aspectRatio, caption, component, credit, isAtEnd, size, and src.
- Updates the `Hero` prop `image` to `imageAlt` and `imageSrc`. Internally, an `Image` component is created.
- Renames the `Image` component props `imageAspectRatio` to `aspectRatio`, `imageCaption` to `caption`, and `imageCredit` to `credit`.
- Updates the `StructuredContent` image-related props into one main prop named `imageProps`. This new prop contains the following properties: alt, aspectRatio, caption, component, credit, position, size, and src.
- Renames the `ToggleSizes.tsx` file to `ToggleTypes.tsx`. Updates the values from `Large` and `Small` to `Default` and `Small`.
- Minor update to the logic for the `ProgressIndicator` sizing prop and styles.
```

Note: In Storybook for `Card` and `StructuredContent`, you'll now see and be able to update `imageProps.alt`, for example, instead of `imageAlt`. It looks odd but I think it's clear that that `imageProps` is an object and `alt` is the property for it that the Controls is updating/modifying. An extra step must be taken in the Story's args though but it's pretty clear, I think, from the Story file.

EdwinGuzman commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr892-tfew8tb9ymfbizvyvlanhovaj4toozuc.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/622683f501c6d20b85350e27

bigfishdesign13 commented on src/components/Card/Card.tsx:
```diff
@@ -7,25 +7,48 @@ import {
   mainActionLink?: string;
 }
 
+// Used internally only for the `imageProps` prop for the `Card` component.
```
Can an interface be setup as a global reusable object?

EdwinGuzman replied:
Kind of. I'm pulling in `ImageProps` for the `CardImage` component. Maybe the naming could be slightly better.

The other place something like this happens in the `StructuredContent` but the only difference is this has `isAtEnd` and `StructuredContent` has `position`. Otherwise, I can update this.

bigfishdesign13 commented on src/components/Hero/Hero.tsx:
```diff
@@ -65,38 +70,44 @@ export default function Hero(props: React.PropsWithChildren<HeroProps>) {
     if (!backgroundImageSrc) {
       console.warn(
-        `Warning: it is recommended to use the "backgroundImageSrc" prop for PRIMARY hero.`
+        `Hero: it is recommended to use the "backgroundImageSrc" prop for the PRIMARY Hero.`
```
Do you think we should be more explicit with the labelling on the warning messages?
```
NYPL Reservoir Hero Component: …
```
Doing so would impact a lot of other components as well.

EdwinGuzman replied:
I think that's too long. Maybe `NYPL Hero: ...` ? Yeah, this may need another ticket for consistency everywhere.

EdwinGuzman replied:
https://jira.nypl.org/browse/DSD-854

bigfishdesign13 commented on src/components/Hero/Hero.tsx:
```diff
@@ -65,38 +70,44 @@ export default function Hero(props: React.PropsWithChildren<HeroProps>) {
       );
     }
-    if (image) {
+    if (imageAlt && imageSrc) {
```
I think we should also have a test to make sure when one of these props is set the other is also set.

bigfishdesign13 commented on src/theme/components/icon.ts:
```diff
@@ -40,6 +40,10 @@ const size = {
     height: "var(--nypl-space-m)",
     width: "var(--nypl-space-m)",
   },
+  default: {
```
The `default` size should go to 100% width.  That is how `Image` and `Logo` treat `default`.

bigfishdesign13 replied:
Sorry, I forgot I had this in here.  This is obviously a requested change.  Does the thumbs-up mean you will make that change in this PR?

EdwinGuzman replied:
Yes, that I'll make a change. For this one, this is width 100%:
<img width="1028" alt="Screen Shot 2022-03-10 at 12 55 51 PM" src="https://user-images.githubusercontent.com/1280564/157737762-13a35437-2e30-4aaa-b31d-85f24cf43fbc.png">

and this is width and height 100%:
<img width="1042" alt="Screen Shot 2022-03-10 at 12 55 42 PM" src="https://user-images.githubusercontent.com/1280564/157737817-01520ac3-79ef-48d7-9bd5-780a451bf2fd.png">

It didn't seem right to me so that's why I made it similar to `medium`. Which one should it be?

EdwinGuzman replied:
I'm making it `width: "auto"`. It is very interesting that it's so similar to `Logo` but that one has `maxWidth` and `Icon` has `width`. Switching to maxWidth doesn't seem to work though....

bigfishdesign13 commented on src/components/StructuredContent/StructuredContent.stories.mdx:
```diff
@@ -111,16 +109,36 @@ export const imageSizeEnumValues = getStorybookEnumValues(
+      className: undefined,
+      headingText: "Heading text",
+      id: "structured-content",
+      "imageProps.alt": "Image alt text",
```
Why are the `imageProps` props in quotes?

EdwinGuzman replied:
It's in quotes so it appears in Storybook's args table and so that the Controls can pick up the changes. Likewise, when getting this value from the `args` object, you have to do `args["imageProps.alt"]` -- which is pretty dumb. I expected `args.imageProps.alt` but this doesn't work in Storybook and others also submitted issues regarding nested values. The default way storybook does this isn't very intuitive so by making it `"imageProps.alt"`, it's hopefully clear that you are updating the `alt` property of the `imageProps` prop object.

bigfishdesign13 replied:
Got it.  Thanks.  And yes, I thought what was showing in the ArgsTable was clear.

bigfishdesign13 reviewed (commented):
Phew!  Wow, that was quite a PR.  I have a few questions sprinkled throughout.

One thing that I would like to see is more consistency with how the sections and info are handled on the stories for `Image`, `Icon` and `Logo`.  For example, the `Size` section on each story is a bit different.  Do you want to attack those changes in this PR or setup a new ticket for that?

EdwinGuzman committed:
Merging and fixing conflicts

bigfishdesign13 commented:
> `Heading` has its own logic which includes different h* sizes, so that one may not work or should be refactored.

Right, this is a tricky one.  I think it is hard to identify “default" styles for a heading, so let’s leave that one alone, at least for now.

oliviawongnyc commented on src/components/Hero/Hero.stories.mdx:
```diff
@@ -39,7 +38,8 @@ export const enumValues = getStorybookEnumValues(HeroTypes, "HeroTypes");
       options: enumValues.options,
     },
-    image: { control: false },
+    imageAlt: { control: false },
```
Would it make sense to use one prop, `imageProps`,  here like you did in the Card component?

EdwinGuzman replied:
Yep! But this is going to be another ticket/PR because the `Hero` component needs to be refactored: https://jira.nypl.org/browse/DSD-614

oliviawongnyc commented on src/components/Image/Image.stories.mdx:
```diff
@@ -98,22 +98,22 @@ export const imageBlockStyles = {
-      imageCredit: "Image credit",
-      imageSize: "ImageSizes.Medium",
+      credit: "Image credit",
       imageType: "ImageTypes.Default",
```
Wondering why we removed the `image` prefix from all but type?

EdwinGuzman replied:
Ah, because this `imageType` is related to the Chakra "variant" for its style. We were thinking of using `type` but the problem is that `type` is an actual HTML attribute and that will be confusing. For example, the HTML button element can take a `type` attribute and it can be set to "submit", "button", or "reset" `<button type="submit">Submit</button>`. To be consistent, the prop to change the variant is `buttonType`, while there also is a separate `type` prop for "submit", "button", and "reset" `<Button buttonType={ButtonTypes.Secondary} type="submit">Submit</Button>`.

EdwinGuzman committed:
Updating based on feedback

EdwinGuzman committed:
Merging and fixing conflicts

oliviawongnyc reviewed (approved):
Wow! A lot of work here. Looks like a big and thoughtful improvement. Excited to implement in the test app.

EdwinGuzman committed:
Making IconSizes Default's width set to auto

EdwinGuzman commented:
@bigfishdesign13 updated the width of the `Icon`.

bigfishdesign13 committed:
cleaned up stories for Icon, Image and Logo

EdwinGuzman commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr892-tfew8tb9ymfbizvyvlanhovaj4toozuc.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/622683f501c6d20b85350e27

EdwinGuzman commented:
@bigfishdesign13 the last update looks good but can you add for the `Icon` stories that you used a specific size for each section? For example, in the Colors section, it'd be good to point out that the size for the color examples are all `IconSizes.ExtraExtraLarge`.

bigfishdesign13 committed:
polishing stories for Icon and Logo

EdwinGuzman committed:
Merging from development and fixing conflicts

EdwinGuzman committed:
Merge branch 'DSD-834/consistent-prop-names' of github.com:NYPL/nypl-design-system into DSD-834/consistent-prop-names

* 'DSD-834/consistent-prop-names' of github.com:NYPL/nypl-design-system:
  polishing stories for Icon and Logo

bigfishdesign13 commented:
> @bigfishdesign13 the last update looks good but can you add for the `Icon` stories that you used a specific size for each section? For example, in the Colors section, it'd be good to point out that the size for the color examples are all `IconSizes.ExtraExtraLarge`.

@EdwinGuzman Thanks for the suggestions.  Please see the updates for `Icon` and `Logo`.

bigfishdesign13 commented:
@EdwinGuzman The `conflicts` are causing the Tugboat build to fail, right?

EdwinGuzman committed:
Merging from development and fixing conflicts

EdwinGuzman committed:
github down

EdwinGuzman committed:
Merging from development and fixing conflicts

EdwinGuzman committed:
Updating Breadcrumb prop name

EdwinGuzman commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr892-tfew8tb9ymfbizvyvlanhovaj4toozuc.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/622683f501c6d20b85350e27

EdwinGuzman committed:
Merge branch 'development' into DSD-834/consistent-prop-names

* development:
  fixed typo

EdwinGuzman commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr892-tfew8tb9ymfbizvyvlanhovaj4toozuc.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/622683f501c6d20b85350e27

EdwinGuzman committed:
Merging from development and fixing conflicts

EdwinGuzman commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr892-tfew8tb9ymfbizvyvlanhovaj4toozuc.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/622683f501c6d20b85350e27

bigfishdesign13 commented:
@EdwinGuzman This looks like it is ready to go.  Is there anything else you wanted to do before merging?

EdwinGuzman commented:
Not sure. I'll update and merge!

EdwinGuzman committed:
Merging from development and fixing conflicts

EdwinGuzman committed:
Minor fixes to the updates

EdwinGuzman commented:
I'm going to wait until @oliviawongnyc 's latest PR on the Accordion is merged in.

EdwinGuzman committed:
Merging and fixing conflicts

EdwinGuzman commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr892-tfew8tb9ymfbizvyvlanhovaj4toozuc.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/622683f501c6d20b85350e27

EdwinGuzman commented:
Going to merge this in.
